### ISSUE TITLE
[1.5.6-phetio] Precipitation particles are failing to be saved in state

### ISSUE THREAD

**Issue by phet-steele** [phetsims/beers-law-lab#228]
@samreid I could have sworn there was an issue about this somewhere at some point, so I think this is a regression and sorry in advance for the duplication.

In BLL 1.5.6-phetio instance-proxies:

1. Concentration screen
2. Shake some particles
3. Evaporate until there is precipitation
4. Press "Launch"

The launched sim will not have any precipitation and will throw this error:

```
TypeError: Cannot read property 'constructor' of undefined
    at Object.getType (beers-law-lab_en-phetio.html?sim=beers-law-lab&phet-io.emitStates=false&phet-io.emitInputEvents=false&sim=beers-law-lab:979)
    at Object.A [as setState] (beers-law-lab_en-phetio.html?sim=beers-law-lab&phet-io.emitStates=false&phet-io.emitInputEvents=false&sim=beers-law-lab:979)
    at TPhETIO.implementation [as setState] (beers-law-lab_en-phetio.html?sim=beers-law-lab&phet-io.emitStates=false&phet-io.emitInputEvents=false&sim=beers-law-lab:1012)
    at Object.invoke (beers-law-lab_en-phetio.html?sim=beers-law-lab&phet-io.emitStates=false&phet-io.emitInputEvents=false&sim=beers-law-lab:1011)
    at Object.handleRequest (beers-law-lab_en-phetio.html?sim=beers-law-lab&phet-io.emitStates=false&phet-io.emitInputEvents=false&sim=beers-law-lab:1011)
    at beers-law-lab_en-phetio.html?sim=beers-law-lab&phet-io.emitStates=false&phet-io.emitInputEvents=false&sim=beers-law-lab:1011
```

The State wrapper is also revealing, but confusing. Run the State wrapper and shake some particles and you'll see what I mean.

And here's my little checklist table thing:

| | |
| --- | --- |
| Problem in 1.5.5-phetio? | Yes |
| Problem in master? | No, but master isn't perfect either |

Seen on macOS 10.13.4 Chrome. For phetsims/QA/issues/128.

**Comment by phet-steele** [comment]
> The State wrapper is also revealing, but confusing. Run the State wrapper and shake some particles and you'll see what I mean.

Details at phetsims/phet-io/issues/1330.

**Comment by zepumph** [comment]
If the problem is in the most recent production version ( 1.5.5-phetio) then we will likely not fix this for the maintenance release.  PhET-iO is in beta, and it was definitely in beta when 1.5.5-phetio was released, so since we haven't heard complaints from partners about this bug. I think the goal is to focus energy on moving phet-io forward. @samreid what do you think?

**Comment by JRomero0613** [comment]
I have noticed that along with the particles failing to be saved by the state that when in the state wrapper the PhET menu is not copied to the smaller simulation. This issue is present in version 1.5.5, 1.5.6 both RC and Production, 1.6.4, and master. For versions 1.5.5, 1.5.6, and 1.6.4 the copied screen does not dim as if the menu is popping up, it does in master which is where the attached screenshot comes from.
![bll_state_phet_menu](https://user-images.githubusercontent.com/30352536/41174820-66fc6a38-6b18-11e8-8371-a17c219b3869.JPG)

**Comment by phet-steele** [comment]
> I have noticed that along with the particles failing to be saved by the state that when in the state wrapper the PhET menu is not copied to the smaller simulation.

@JRomero0613 great that you noticed, but this is expected in the State wrapper (although it would be a problem in the Mirror Inputs wrapper).

**Comment by samreid** [comment]
> but this is expected in the State wrapper (although it would be a problem in the Mirror Inputs wrapper).

To elaborate, we decided this is a "transient" behavior that should not be saved/loaded.

**Comment by JRomero0613** [comment]
Okay, I have just checked all versions mentioned above in the mirror inputs wrapper and they all function properly

**Comment by zepumph** [comment]
Can this issue be closed then?

**Comment by phet-steele** [comment]
> Can this issue be closed then?

No. It's still an issue, even if it does not get fixed right away.

**Comment by samreid** [comment]
I reproduced the problem in 1.5.6-phetio, not planning on solving it in the branch, unless I hear otherwise.  Good to know though, in case our clients run into it.  Unassigning for now.

**Comment by pixelzoom** [comment]
The issue of what precipitate state to save/restore will be addressed in #230 (redesign PhET-iO API). 
So I doubt that we will ever address this in the 1.5-phetio branch.  @samreid can this issue be closed with "won't fix" ?
